lib/bundler/resolver.rb

```diff
@@ -124,23 +124,17 @@ def source
 
       def for?(platform, ruby_version)
         spec = @specs[platform]
-        return false unless spec
-
-        return true if ruby_version.nil?
-        # Only allow endpoint specifications since they won't hit the network to
-        # fetch the full gemspec when calling required_ruby_version
-        return true if !spec.is_a?(EndpointSpecification) && !spec.is_a?(Gem::Specification)
-        return true if spec.required_ruby_version.nil?
-
-        spec.required_ruby_version.satisfied_by?(ruby_version.to_gem_version_with_patchlevel)
+        !spec.nil?
```

Comment:
Sorry, I was kind of vague here.

Instead of this:

```ruby
def for(platform)
  spec = @specs[platform]
  !spec.nil?
end
```

How about this?

```ruby
def for?(platform)
  @specs[platform]
end
```

In the end I don't care enough to block on this, though.